api/src/main/java/io/strimzi/api/kafka/model/listener/v2/ArrayOrObjectKafkaListeners.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener.v2;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.listener.KafkaListeners;
+import io.strimzi.crdgenerator.annotations.Alternation;
+import io.strimzi.crdgenerator.annotations.Alternative;
+import io.sundr.builder.annotations.Buildable;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonDeserialize(using = ArrayOrObjectKafkaListeners.Deserializer.class)
+@JsonSerialize(using = ArrayOrObjectKafkaListeners.Serializer.class)
+@Alternation
+public class ArrayOrObjectKafkaListeners implements Serializable {
```

Comment:
Aren't arrays Objects too?  I know this is petty, but the fact that the new KafkaListeners handles objects or arrays is surely an implementations details (and therefore should be doc commented or made clear through the flow of the code?

Reply:
While it's true that in Java arrays are reference types, this is about the JSON schema and in JSON array and object are considered separate types. To me the name makes sense.

Reply:
Well, in YAML this is very different. But if you propose a better name I'm fine to use it. I think this describes the two form fairly well, but I agree it is a bit weird name.

Reply:
Ah fair, I hadn't thought through the meaning in JSON.
Given this is in the API, I agree this name names sense as it is a serializer.